src/Compilers/Core/Portable/Emit/EmitOptions.cs

```diff
@@ -177,6 +178,14 @@ public bool Equals(EmitOptions other)
                 return false;
             }
 
+            for (int i = 0; i < this.InstrumentationKinds.Length; i++)
+            {
+                if (this.InstrumentationKinds[i] != other.InstrumentationKinds[i])
+                {
+                    return false;
```

Comment:
It looks like order matters in this equality comparison. Can you confirm this is intentional?